src/Symfony/Component/Form/Tests/Extension/Core/Type/CheckboxTypeTest.php

```diff
@@ -170,4 +170,38 @@ public function provideCustomModelTransformerData()
             array('unchecked', false),
         );
     }
+
+    public function testSubmitNullWithEmptyDataTrueWhenNested()
+    {
+        $form = $this->factory->create('form')
```

Comment:
Ok, actually @webmozart is right.

You can remove "WhenNested" is the method name and refractor this to:
```php
$form = $this->factory->create('checkbox', null, array(
                 'empty_data' => true,
             ));
 
         $form->submit(null);
 
         $this->assertTrue($form->getData());
         $this->assertSame('1', $form->getViewData());